src/aggregation/metric/top_hits.rs

```diff
@@ -0,0 +1,827 @@
+                ),
+            },
+        ];
+
+        let mut collector = collector_with_capacity(3);
+        for doc in docs.clone() {
+            collector.collect(doc.feature, doc.doc);
+        }
+
+        let res = collector.finalize();
+
+        assert_eq!(
+            res,
+            super::TopHitsMetricResult {
+                hits: vec![
+                    super::TopHitsVecEntry {
+                        id: docs[0].doc,
+                        sort: vec![docs[0].feature.0[0].value],
+                        search_results: Default::default(),
+                    },
+                    super::TopHitsVecEntry {
+                        id: docs[1].doc,
+                        sort: vec![docs[1].feature.0[0].value],
+                        search_results: Default::default(),
+                    },
+                    super::TopHitsVecEntry {
+                        id: docs[2].doc,
+                        sort: vec![docs[2].feature.0[0].value],
+                        search_results: Default::default(),
+                    },
+                ]
+            }
+        );
+
+        Ok(())
+    }
+
+    fn test_aggregation_top_hits(merge_segments: bool) -> crate::Result<()> {
+        let docs = vec![
+            vec![
+                r#"{ "date": "2015-01-02T00:00:00Z", "text": "bbb", "text2": "bbb" }"#,
+                r#"{ "date": "2017-06-15T00:00:00Z", "text": "ccc", "text2": "ddd" }"#,
+            ],
+            vec![
+                r#"{ "text": "aaa", "text2": "bbb", "date": "2018-01-02T00:00:00Z" }"#,
+                r#"{ "text": "aaa", "text2": "bbb", "date": "2016-01-02T00:00:00Z" }"#,
+            ],
+        ];
+
+        let index = get_test_index_from_docs(merge_segments, &docs)?;
+
+        let d: Aggregations = serde_json::from_value(json!({
+            "top_hits_req": {
+                "top_hits": {
+                    "size": 2,
+                    "sort": [
+                        { "date": "desc" }
+                    ],
+                    "from": 1,
+                    "docvalue_fields": [
+                        "date"
+                    ],
+                }
+        }
+        }))?;
+
+        let collector = AggregationCollector::from_aggs(d, Default::default());
+        let reader = index.reader()?;
+        let searcher = reader.searcher();
+
+        let agg_res: AggregationResults = searcher.search(&AllQuery, &collector).unwrap();
+
+        let date_2017 = datetime!(2017-06-15 00:00:00 UTC);
+        let date_2016 = datetime!(2016-01-02 00:00:00 UTC);
+
+        assert_eq!(
+            agg_res
+                .0
+                .get("top_hits_req")
+                .expect("top_hits_req must exist"),
+            &AggregationResult::MetricResult(MetricResult::TopHits(TopHitsMetricResult {
+                hits: vec![
+                    TopHitsVecEntry {
+                        id: crate::DocAddress {
+                            segment_ord: 0,
```

Comment:
The merge_segments flag merges the segments,so `segment_ord` depends on that flag

Reply:
I've already accounted for that condition. Now it appears that the ordinal of the segments is non-determinate from their declaration order in the initialization vector: so the segment declared [first](https://github.com/quickwit-oss/tantivy/pull/2198/files#diff-da09c2e16f884b4f3436a59897de837ff6b5b49003f54ed3e7d9e6687d7229d6R731-R734) might have ord `1` instead of the expected `0` when `merge_segments` is `false`. My initial hypothesis was a race condition between segment writers but that doesn't seem to be the case here.

Reply:
Resolving since top_hits' tests don't have reliance on addresses anymore, but this is still potentially an issue for other tests in the future.